host/cli/volume.go

```diff
@@ -0,0 +1,210 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/flynn/flynn/host/volume"
+	"github.com/flynn/flynn/pkg/cluster"
+	"github.com/flynn/go-docopt"
+)
+
+func init() {
+	Register("volume", runVolume, `
+usage: flynn-host volume list
+       flynn-host volume create [--provider] HOST
+       flynn-host volume delete ID...
+       flynn-host volume gc
+
+Commands:
+    list    Display a list of all volumes of known Flynn hosts
+    create  Creates a volume on a host
+    delete  Deletes volumes, destroying any data stored on them
+    gc      Garbage collect currently unused volumes
+
+Examples:
+
+    $ flynn-host volume list
+
+    $ flynn-host volume create --provider zfs host0
+
+    $ flynn-host volume destroy 102fad07-07a3-4841-bded-d9e8a3eedbd6
+
+    $ flynn-host volume gc
+`)
+}
+
+func runVolume(args *docopt.Args, client *cluster.Client) error {
+	switch {
+	case args.Bool["list"]:
+		return runVolumeList(args, client)
+	case args.Bool["delete"]:
+		return runVolumeDelete(args, client)
+	case args.Bool["create"]:
+		return runVolumeCreate(args, client)
+	case args.Bool["gc"]:
+		return runVolumeGarbageCollection(args, client)
+	}
+	return nil
+}
+
+func runVolumeGarbageCollection(args *docopt.Args, client *cluster.Client) error {
+	// collect list of all volume ids currently attached to jobs
+	hosts, err := client.Hosts()
+	if err != nil {
+		return fmt.Errorf("could not list hosts: %s", err)
+	}
+	if len(hosts) == 0 {
+		return errors.New("no hosts found")
+	}
+
+	attached := make([]string, 0, 0)
+	for _, h := range hosts {
+		jobs, err := h.ListJobs()
+		if err != nil {
+			return err
```

Comment:
I think we should just print this error and move on so one host in a bad state doesn't prevent all volume GC.